code/modules/mob/living/silicon/robot/robot_damage.dm

```diff
@@ -107,9 +107,9 @@
 	if(status_flags & GODMODE)
 		return
 
-	if(damage_protection)
-		brute = clamp(brute - damage_protection, 0, brute)
-		burn = clamp(burn - damage_protection, 0, burn)
+	if(damage_protection || burn_mod != 1 || brute_mod != 1)
+		brute = clamp((brute - damage_protection)*brute_mod, 0, brute)
+		burn = clamp((burn - damage_protection)*burn_mod, 0, burn)
```

Comment:
I think it would better to just remove the if statement here and have the code below it run always. It's not expensive and borgs taking damage doesn't happen that many times during a round. And I think the expense of checking 3 conditions is probably close to whatever expense the code under it costs anyway.

Reply:
Sure, one moment